cmd/crates/soroban-spec-typescript/src/boilerplate.rs

```diff
@@ -74,29 +77,48 @@ impl Project {
             ("INSERT_RPC_URL_HERE", rpc_url),
         ];
         let root: &Path = self.as_ref();
-        [
-            "package.json",
-            "README.md",
-            "src/constants.ts",
-            "src/index.ts",
-            "src/invoke.ts",
-        ]
-        .into_iter()
-        .try_for_each(|file_name| {
-            let file = &root.join(file_name);
-            let mut contents = fs::read_to_string(file).unwrap();
-            for (pattern, replacement) in replacement_strings {
-                contents = contents.replace(pattern, replacement);
-            }
-            fs::write(file, contents)
-        })
+        ["package.json", "README.md", "src/index.ts", "src/invoke.ts"]
+            .into_iter()
+            .try_for_each(|file_name| {
+                let file = &root.join(file_name);
+                let mut contents = fs::read_to_string(file).unwrap();
+                for (pattern, replacement) in replacement_strings {
+                    contents = contents.replace(pattern, replacement);
+                }
+                fs::write(file, contents)
+            })
     }
 
-    fn append_index_ts(&self, spec: &[ScSpecEntry]) -> std::io::Result<()> {
+    fn append_index_ts(
+        &self,
+        spec: &[ScSpecEntry],
+        contract_id: &str,
+        network_passphrase: &str,
+    ) -> std::io::Result<()> {
+        let networks = Project::format_networks_object(contract_id, network_passphrase);
+        let types_and_fns = generate(spec);
         fs::OpenOptions::new()
             .append(true)
             .open(self.0.join("src/index.ts"))?
-            .write_all(generate(spec).as_bytes())
+            .write_all(format!("\n\n{networks}\n\n{types_and_fns}").as_bytes())
+    }
+
+    fn format_networks_object(contract_id: &str, network_passphrase: &str) -> String {
+        let network = if network_passphrase == NETWORK_PASSPHRASE_FUTURENET {
+            "futurenet"
+        } else if network_passphrase == NETWORK_PASSPHRASE_LOCALNET {
+            "localnet"
+        } else {
+            "unknown"
+        };
+        format!(
+            r#"export const networks = {{
+    {network}: {{
+        networkPassphrase: "{network_passphrase}",
```

Comment:
you can use e.g. `SorobanClient.Networks.FUTURENET` if you don't want the magic strings repeated

Reply:
Good to know! I think the magic string makes more sense in this context, to reduce the reliance on dependencies in the generated client.

Reply:
We'll also want to support pubnet and testnet soon, so might be worth doing them as well.

I had expected the  network passphrase & url would come from the `soroban config network` setup the user would have already done.  So if I did:
```bash
$ soroban config network add \
    --name chocolate_factory \
    --passphrase "Oompa Loompa Network ; October 2022" \
    --rpc-url "https://charlie-rpc.com"
$ soroban contract binding typescript --network chocolate_factory --id C1234
```
then it would use the network config settings from that network to generate a network setup like:
```ts
export const networks = {
  readonly "chocolate_factory": {
    // whatever format the client expects so you could pass it directly
    readonly networkPassphrase: "Oompa Loompa Network ; October 2022",
    readonly rpcUrl: "https://charlie-rpc.com",
    readonly contractId: "C1234",
  },
};
```

Reply:
> to reduce the reliance on dependencies in the generated client

but there's no world in which you can do this without SorobanClient, I think?